src/textual/css/stylesheet.py

```diff
@@ -433,13 +433,14 @@ def _check_rule(
             if _check_selectors(selector_set.selectors, css_path_nodes):
                 yield selector_set.specificity
 
-    # pseudo classes which iterate over many nodes
-    # these have the potential to be slow, and shouldn't be used in a cache key
-    EXPENSIVE_PSEUDO_CLASSES = {
+    # pseudo classes which iterate over multiple nodes
+    # These shouldn't be used in a cache key
+    EXCLUDE_PSEUDO_CLASSES_FROM_CACHE = {
```

Comment:
Does this need to be a classvar if it's never going to change?

Reply:
Good point. I think actually `Final` would be appropriate, and making it private. Since its not something the dev should know / care about / change.